### ISSUE TITLE
Dictionary citation not showing selected page

### ISSUE THREAD

**Issue by lyndondrake** [dcpurton/biblatex-sbl#122]
I have the below BibLaTeX entries for ABD and an article in it. I have the following citation:

```
\autocite[1002]{ABD:writing}
```

<img width="347" alt="Screenshot 2021-06-30 at 11 29 05 AM" src="https://user-images.githubusercontent.com/18509357/123880268-65bae100-d996-11eb-9b2d-e4eee7bd5a94.png">

The expected output for this would be:

Andreé Lemaire, 'Writing and Writing Materials', *ABD* 6:1002.

Similar to the IBD example on p. 22–23 of the biblatex-sbl-examples PDF. The shorthand isn't being used, and instead of the individual page the whole entry page range is being shown.

Any idea what I'm doing wrong? The bib file is generated by Zotero BTW.

```
@mvreference{ABD,
  title = {The {{Anchor Yale Bible Dictionary}}},
  shorttitle = {{{ABD}}},
  editor = {Freedman, David Noel and Herion, Gary A and Graf, David F and Beck, Astrid B},
  date = {1990},
  publisher = {{Doubleday}},
  location = {{New York}},
  shorthand = {ABD},
  volumes = {6}
}

@inreference{ABD:writing,
  title = {Writing and Writing Materials},
  booktitle = {The {{Anchor Yale Bible Dictionary}}},
  author = {Lemaire, Andreé},
  editor = {Freedman, David Noel and Herion, Gary A and Graf, David F and Beck, Astrid B},
  date = {1990},
  volume = {6},
  pages = {999--1009},
  publisher = {{Doubleday}},
  location = {{New York}},
  crossref = {ABD},
  volumes = {6},
  xref = {ABD}
}
```

**Comment by dcpurton** [comment]
This was by design. The handbook is super unclear on when/if full page ranges should be given for the first citation.

I think I ended up always putting a full page range in for the first citation if there is `pages` field in the bib entry.

With further [clarification from the Blog](https://sblhs2.com/2017/04/13/citing-reference-works-5-topical-dictionaries-and-encyclopedias/) I think this is wrong and for reference articles only relevant page(s) should be included in the first citation. Full page range only needs to go in the bibliography.

I've changed this in my near implementation.

I'll see if I can provide a workaround for you.

**Comment by lyndondrake** [comment]
I've just read that blog, which still doesn't *precisely* address the question, but I agree that 3.1 and 3.2 imply that the first citation if to a specific page should only reference the relevant page(s), with the full page range in the bibliography.

I don't have an urgent need for it to be fixed, as I'm still several weeks away from submitting the relevant piece of work for any formal assessment, so I can either test a release for you if that would help, or wait for an update (I'm using TeXLive).

**Comment by dcpurton** [comment]
It seems like SBL's current recommendation is that citations never include full page ranges when there is a bibliography containing the full page range.

See https://sblhs2.com/2018/02/08/citing-page-numbers-for-chapters-and-articles/

This behaviour can be activated using the `citepages=omit` package option.

Not that this affects all entry types with a `pages` field.

Will this do what you want?

**Comment by dcpurton** [comment]
Perhaps I should make `citepages=omit` the default.

Although this would break existing documents that don't expect this behaviour.

**Comment by lyndondrake** [comment]
> This behaviour can be activated using the citepages=omit package option.

That does indeed seem to do the trick. I'm happy using that.

BTW I am very grateful for your work on `sbl-biblatex`, not least because of the ambiguities and oddities involved!

**Comment by folofjc** [comment]
> Perhaps I should make `citepages=omit` the default.
> 
> Although this would break existing documents that don't expect this behaviour.

I would agree that this should be the default. If you are putting this in your new implementation, perhaps it will not break too many existing documents since they may not update to the new one. 

BTW, I also include the `citepages=omit` package option in all my docs.

**Comment by dcpurton** [comment]
I already have switched to `citpages=omit` in my new code. Which I think I will put up on GitHub soon!

**Comment by dcpurton** [comment]
@lyndondrake and @folofjc I just committed my new code to the v2 branch.

There's no documentation and it isn't backward compatible, so don't try and build your current document with it. It's also definitely *not* ready yet.

Also, I didn't change the name (!), which is probably and error, so it's still loaded with with `style=sbl`.

If you want to test it, don't install, just copy the files to the directory where your TeX file is.

You can see how things are done in the files in `testfiles` and in `biblatex-sbl.bib`.

**Comment by folofjc** [comment]
Thanks! I'll try to give it a test. Do you have any suggestions on how to have both this and the old one in the same TeX installation? I don't want to break the documents I am working on.

> Also, I didn't change the name (!), which is probably and error, so it's still loaded with with `style=sbl`

Are you planning on renaming it and keeping the old version around when you put it on CTAN? That might be confusing...

**Comment by dcpurton** [comment]
> Thanks! I'll try to give it a test. Do you have any suggestions on how to have both this and the old one in the same TeX installation? I don't want to break the documents I am working on.

I think it best not to install. Just symlink (or copy if you are stuck with Windows) the files to the same directory as your test TeX file.

> 
> > Also, I didn't change the name (!), which is probably and error, so it's still loaded with with `style=sbl`
> 
> Are you planning on renaming it and keeping the old version around when you put it on CTAN? That might be confusing...

I haven't decided what I will do with renaming. I may rename the old one and keep one package or go with `sbl2` for the new one with (maybe) a completely different package.

Happy to take votes.

**Comment by lyndondrake** [comment]
It might be the wrong place to ask, but I wondered where v2 is up to?

**Comment by dcpurton** [comment]
It's the right place to ask!

It's been neglected a bit, but I'll get back to it. I have some long service leave coming up in the middle of the year and will see if I can do some more.

**Comment by folofjc** [comment]
> Thanks! I'll try to give it a test. Do you have any suggestions on how to have both this and the old one in the same TeX installation? I don't want to break the documents I am working on.

Yikes, I said I was going to test this a long time ago. Whoops. So I have also been neglecting!
